segfault while using karmor recommend

karmor version: 0.9.5

```
INFO[0045] dumped image to tar                           tar=/tmp/karmor2162624332/wbdoJOYh.tar
created policy out/kubernetes-dashboard-kubernetes-dashboard/kubernetesui-dashboard-v2.6.1-password-protect.yaml ...
INFO[0046] pulling image                                 image="quay.io/cilium/operator-generic:v1.11.3@sha256:5b81db7a32cb7e2d00bb3cf332277ec2b3be239d9e94a8d979915f4e6648c787"
quay.io/cilium/operator-generic@sha256:5b81db7a32cb7e2d00bb3cf332277ec2b3be239d9e94a8d979915f4e6648c787: Pulling from cilium/operator-generic
bc877eec10d7: Pull complete 
78ea17f4e2e5: Pull complete 
508c65bb69fc: Pull complete 
34887728791f: Pull complete 
Digest: sha256:5b81db7a32cb7e2d00bb3cf332277ec2b3be239d9e94a8d979915f4e6648c787
Status: Downloaded newer image for quay.io/cilium/operator-generic@sha256:5b81db7a32cb7e2d00bb3cf332277ec2b3be239d9e94a8d979915f4e6648c787
INFO[0055] dumped image to tar                           tar=/tmp/karmor1431945542/JspJTTYI.tar
panic: interface conversion: interface {} is nil, not []interface {}

goroutine 1 [running]:
github.com/kubearmor/kubearmor-client/recommend.(*ImageInfo).readManifest(0xc00090c210, {0xc000834000?, 0xc000212b00?})
	/home/runner/work/kubearmor-client/kubearmor-client/recommend/imageHandler.go:280 +0x838
github.com/kubearmor/kubearmor-client/recommend.(*ImageInfo).getImageInfo(0xc00090c210)
	/home/runner/work/kubearmor-client/kubearmor-client/recommend/imageHandler.go:387 +0x22d
github.com/kubearmor/kubearmor-client/recommend.getImageDetails({0xc0004554b0, 0xb}, {0xc000455490, 0xf}, 0xc000161ad0, {0xc000d79b90, 0x6f})
	/home/runner/work/kubearmor-client/kubearmor-client/recommend/imageHandler.go:407 +0x1cd
github.com/kubearmor/kubearmor-client/recommend.imageHandler({0xc0004554b0, 0xb}, {0xc000455490, 0xf}, 0x4172d3?, {0xc000d79b90, 0x6f})
	/home/runner/work/kubearmor-client/kubearmor-client/recommend/imageHandler.go:423 +0x1ae
github.com/kubearmor/kubearmor-client/recommend.handleDeployment({{0xc000455490, 0xf}, {0xc0004554b0, 0xb}, 0xc000161ad0, {0xc00040ffb0, 0x1, 0x1}})
	/home/runner/work/kubearmor-client/kubearmor-client/recommend/recommend.go:153 +0x1a6
github.com/kubearmor/kubearmor-client/recommend.Recommend(0xc0005a7400, {{0x3071230, 0x0, 0x0}, {0x3071230, 0x0, 0x0}, {0x0, 0x0, 0x0}, ...})
	/home/runner/work/kubearmor-client/kubearmor-client/recommend/recommend.go:136 +0x3fc
github.com/kubearmor/kubearmor-client/cmd.glob..func11(0x301c380?, {0x1de561f?, 0x0?, 0x0?})
	/home/runner/work/kubearmor-client/kubearmor-client/cmd/recommend.go:19 +0x58
github.com/spf13/cobra.(*Command).execute(0x301c380, {0x3071230, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0x301c600)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/kubearmor/kubearmor-client/cmd.Execute()
	/home/runner/work/kubearmor-client/kubearmor-client/cmd/root.go:49 +0x25
main.main()
	/home/runner/work/kubearmor-client/kubearmor-client/main.go:10 +0x17
```

--- a/Makefile
+++ b/Makefile
@@ -15,6 +15,10 @@ GIT_INFO := $(shell govvv -flags -pkg $(PKG))
 build:
 	cd $(CURDIR); go mod tidy; CGO_ENABLED=0 go build -ldflags "-w -s ${GIT_INFO}" -o karmor
 
+.PHONY: debug
+debug:
+	cd $(CURDIR); go mod tidy; CGO_ENABLED=0 go build -ldflags "${GIT_INFO}" -o karmor
+
 .PHONY: install
 install: build
 	install -m 0755 karmor $(DESTDIR)$(INSTALLDIR)

--- a/recommend/imageHandler.go
+++ b/recommend/imageHandler.go
@@ -277,8 +277,15 @@ func (img *ImageInfo) readManifest(manifest string) {
 	}
 	img.Arch = cfgres["architecture"].(string)
 	img.OS = cfgres["os"].(string)
-	for _, tag := range man["RepoTags"].([]interface{}) {
-		img.RepoTags = append(img.RepoTags, tag.(string))
+
+	if man["RepoTags"] == nil {
+		// If the image name contains sha256 digest,
+		// then manifest["RepoTags"] will be `nil`.
+		img.RepoTags = append(img.RepoTags, shortenImageNameWithSha256(img.Name))
+	} else {
+		for _, tag := range man["RepoTags"].([]interface{}) {
+			img.RepoTags = append(img.RepoTags, tag.(string))
+		}
 	}
 }
 
@@ -426,4 +433,13 @@ func imageHandler(namespace, deployment string, labels LabelMap, imageName strin
 	}
 
 	return nil
+}
+
+// shortenImageNameWithSha256 truncates the sha256 digest in image name
+func shortenImageNameWithSha256(name string) string {
+	if strings.Contains(name, "@sha256:") {
+		// shorten sha256 to first 8 chars
+		return name[:len(name)-56]
+	}
+	return name
 }

--- a/recommend/policy.go
+++ b/recommend/policy.go
@@ -102,6 +102,8 @@ func mkPathFromTag(tag string) string {
 		"/", "-",
 		":", "-",
 		"\\", "-",
+		".", "-",
+		"@", "-",
 	)
 	return r.Replace(tag)
 }

